packages/nextjs/utils/scaffold-stark/__test__/types.test.ts

```diff
@@ -0,0 +1,211 @@
+    });
+
+    it("should return false for invalid Cairo integer types", () => {
+      expect(isCairoInt("core::integer::u64")).toBe(false);
+      expect(isCairoInt("core::integer::u256")).toBe(false);
+      expect(isCairoInt("core::felt252")).toBe(false);
+      expect(isCairoInt("")).toBe(false);
+    });
+  });
+
+  describe("isCairoBigInt", () => {
+    it("should return true for valid Cairo big integer types", () => {
+      expect(isCairoBigInt("core::integer::u64")).toBe(true);
+      expect(isCairoBigInt("core::integer::u128")).toBe(true);
+      expect(isCairoBigInt("core::integer::i64")).toBe(true);
+      expect(isCairoBigInt("core::integer::i128")).toBe(true);
+    });
+
+    it("should return false for invalid Cairo big integer types", () => {
+      expect(isCairoBigInt("core::integer::u32")).toBe(false);
+      expect(isCairoBigInt("core::integer::u256")).toBe(false);
+      expect(isCairoBigInt("")).toBe(false);
+    });
+  });
+
+  describe("isCairoU256", () => {
+    it("should return true for Cairo u256 type", () => {
+      expect(isCairoU256("core::integer::u256")).toBe(true);
+    });
+
+    it("should return false for non-u256 types", () => {
+      expect(isCairoU256("core::integer::u128")).toBe(false);
+      expect(isCairoU256("")).toBe(false);
+    });
+  });
+
+  describe("isCairoContractAddress", () => {
+    it("should return true for Cairo contract address type", () => {
+      expect(
+        isCairoContractAddress(
+          "core::starknet::contract_address::ContractAddress",
+        ),
+      ).toBe(true);
+    });
+
+    it("should return false for non-contract address types", () => {
+      expect(
+        isCairoContractAddress("core::starknet::eth_address::EthAddress"),
+      ).toBe(false);
+      expect(isCairoContractAddress("")).toBe(false);
+    });
+  });
+
+  describe("isCairoFelt", () => {
+    it("should return true for Cairo felt252 type", () => {
+      expect(isCairoFelt("core::felt252")).toBe(true);
+    });
+
+    it("should return false for non-felt types", () => {
+      expect(isCairoFelt("core::integer::u256")).toBe(false);
+      expect(isCairoFelt("")).toBe(false);
+    });
+  });
+
+  describe("isCairoTuple", () => {
+    it("should return true for Cairo tuple types", () => {
+      expect(isCairoTuple("(core::felt252, core::integer::u256)")).toBe(true);
+      expect(isCairoTuple("(core::felt252)")).toBe(true);
+    });
+
+    it("should return false for empty tuple and non-tuple types", () => {
+      expect(isCairoTuple("()")).toBe(false);
+      expect(isCairoTuple("core::felt252")).toBe(false);
+      expect(isCairoTuple("")).toBe(false);
+    });
+  });
+
+  describe("isStructOrEnum", () => {
+    it("should return true for struct types", () => {
+      expect(isStructOrEnum({ type: "struct" })).toBe(true);
+    });
+
+    it("should return true for enum types", () => {
+      expect(isStructOrEnum({ type: "enum" })).toBe(true);
+    });
```

Comment:
can we have a real looking example, would be better IMO

<img width="535" alt="image" src="https://github.com/user-attachments/assets/e5859047-0865-4545-bc47-ab20d473b17b" />